File: main/cs/Ivy/Event/TouchEvent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Ivy.Event
{
    public class TouchEvent : IvyEvent
    {
        private int x = -1;
        private int y = -1;
        private PadButtonAction state = PadButtonAction.DOWN_STATE;

        public TouchEvent(int playerIdx, int xTouch, int yTouch, int state) : base(playerIdx)
        {
            x = xTouch;
            y = yTouch;
        }

        public int getX()
        {
            return x;
        }

        public int getY()
        {
            return y;
        }

        public PadButtonAction getAction()
        {
            return state;
        }

        public override String ToString()
        {
            StringBuilder str = new StringBuilder();

            str.Append("[TOUCH]");
            str.Append(" PLAYER=");
            str.Append(getPlayer());
            str.Append(", X=");
            str.Append(x);
            str.Append(", Y=");
            str.Append(y);
            str.Append(", STATE=");
            str.Append(stateToString(state));

            return str.ToString();
        }
    }
}

File: main/java/src/event/TouchEvent.java
package event;

public class TouchEvent extends IvyEvent {
	private int x = -1;
	private int y = -1;
	private int state = -1;
	
	public TouchEvent(int playerIdx, int xTouch, int yTouch, int state){
		super(playerIdx);
		//playerIndex = playerIdx;
		x = xTouch;
		y = yTouch;
	}
	
	public int getX(){
		return x;
	}
	
	public int getY(){
		return y;
	}
	
	public int getAction(){
		return state;
	}
	
	public String toString(){
		StringBuffer str = new StringBuffer();
		
		str.append("[TOUCH]");
		str.append(" PLAYER=");
		str.append(getPlayer());
		str.append(", X=");
		str.append(x);
		str.append(", Y=");
		str.append(y);
		str.append(", STATE=");
		str.append(stateToString(state));
		
		return str.toString();
	}
}
